TestON/tests/FuncIntent/Dependency/FuncIntentFunction.py

```diff
@@ -0,0 +1,543 @@
+"""
+    Wrapper functions for FuncIntent
+    This functions include Onosclidriver and Mininetclidriver driver functions
+"""
+def __init__( self ):
+    self.default = ''
+
+def hostIntent( main,
+                name="",
+                host1="",
+                host2="",
+                host1Id="",
+                host2Id="",
+                mac1="",
+                mac2="",
+                vlan1="-1",
+                vlan2="-1",
+                sw1="",
+                sw2="",
+                expectedLink=0 ):
+    """
+        Add host intents
+    """
+    import time
+
+    # Assert variables
+    assert main, "There is no main variable"
+    assert name, "variable name is empty"
+    assert host1 and host2, "You must specify hosts"
+
+    global itemName
+    itemName = name
+    h1Name = host1
+    h2Name = host2
+    h1Id = host1Id
+    h2Id = host2Id
+    h1Mac = mac1
+    h2Mac = mac2
+    vlan1 = vlan1
+    vlan2 = vlan2
+    intentsId = []
+    stepResult = main.TRUE
+    pingResult = main.TRUE
+    intentResult = main.TRUE
+    flowResult = main.TRUE
+    topoResult = main.TRUE
+    linkDownResult = main.TRUE
+    linkUpResult = main.TRUE
+
+    if main.hostsData:
```

Comment:
Yes, so it is mainly optional. if they want to pass just host name eg. h1
h2 etc. it will grab those host data and continue with the test otherwise,
manual data is needed to run the test

On Tue, May 19, 2015 at 11:04 AM, Jon Hall <notifications@github.com> wrote:

> In TestON/tests/FuncIntent/Dependency/FuncIntentFunction.py
> <https://github.com/opennetworkinglab/ONLabTest/pull/117#discussion_r30627883>
> :
>
> > +    h1Id = host1Id
> > +    h2Id = host2Id
> > +    h1Mac = mac1
> > +    h2Mac = mac2
> > +    vlan1 = vlan1
> > +    vlan2 = vlan2
> > +    intentsId = []
> > +    stepResult = main.TRUE
> > +    pingResult = main.TRUE
> > +    intentResult = main.TRUE
> > +    flowResult = main.TRUE
> > +    topoResult = main.TRUE
> > +    linkDownResult = main.TRUE
> > +    linkUpResult = main.TRUE
> > +
> > +    if main.hostsData:
>
> Im curious as to why you are getting this info from hostsData if not
> passed in. Is this just so someone can write a testcase without being
> familiar with the topology?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opennetworkinglab/ONLabTest/pull/117/files#r30627883>.
>